staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go

```diff
@@ -185,18 +210,32 @@ func (reqMgmt *requestManagementSystem) Wait(requestDigest RequestDigest) (execu
 }
 
 func (rmState *RequestManagementState) pickFlowSchema(rd RequestDigest) *rmtypesv1alpha1.FlowSchema {
+	for _, flowSchema := range rmState.flowSchemas {
+		if matchesFlowSchema(rd, flowSchema) {
+			return flowSchema
+		}
+	}
 	return nil
-	// TODO: implement
 }
 
 // ComputeFlowDistinguisher extracts the flow distinguisher according to the given method
 func (rd RequestDigest) ComputeFlowDistinguisher(method *rmtypesv1alpha1.FlowDistinguisherMethod) string {
-	// TODO: implement
-	return ""
+	if method == nil {
+		return ""
+	}
+	switch method.Type {
+	case rmtypesv1alpha1.FlowDistinguisherMethodByUserType:
+		return rd.User.GetName()
+	case rmtypesv1alpha1.FlowDistinguisherMethodByNamespaceType:
+		return rd.RequestInfo.Namespace
+	default:
+		// this line shall never reach
+		panic("invalid flow-distinguisher method")
+	}
 }
 
 // HashFlowID hashes the inputs into 64-bits
 func hashFlowID(fsName, fDistinguisher string) uint64 {
-	// TODO: implement
-	return 0
+	hash := sha256.Sum256([]byte(fsName + fDistinguisher))
+	return big.NewInt(0).SetBytes(hash[:8]).Uint64()
```

Comment:
I think this may be more expensive than we want.  I expect `hash/crc64` will be cheaper and adequate for our needs.

Reply:
updated to crc64